mesonbuild/interpreterbase.py

```diff
@@ -1026,6 +1026,28 @@ def string_method_call(self, obj: str, method_name: str, posargs: T.List[TYPE_nv
             if not isinstance(cmpr, str):
                 raise InterpreterException('Version_compare() argument must be a string.')
             return mesonlib.version_compare(obj, cmpr)
+        elif method_name == 'substring':
+            if len(posargs) > 2:
+                raise InterpreterException('substring() takes maximum two arguments.')
+            start = 0
+            end = len(obj)
+            if len (posargs) > 0:
+              start = int(posargs[0])
+            if len (posargs) > 1:
+                end = int(posargs[1])
+            if start < 0:
+                start = 0
+            if end < 0:
+                if abs(end) <= len(obj):
+                    end = len(obj) - abs(end)
+                else:
+                    end = 0
+            # swap if start > stop
+            if start > end:
+                s = start
+                start = end
+                end = s
```

Comment:
Why swapping? IMHO `'abc'.substring(1, 0)` should return empty string, like python does.

Reply:
I took it from javascript as I should have taken from python ;)

Reply:
ah, didn't know JS does that. I guess it depends on the language you're used to... I'm certainly biased toward python, maybe that's not the behaviour others would expect?